clients/src/main/java/org/apache/kafka/common/protocol/types/Type.java

```diff
@@ -59,6 +59,47 @@ public boolean isNullable() {
         return false;
     }
 
+    /**
+     * The Boolean type represents a boolean value in a byte by using
+     * the value of 0 to represent false, and 1 to represent true.
+     *
+     * If for some reason a value that is not 0 or 1 is read,
+     * then any non-zero value will return true.
```

Comment:
What is the reasoning for accepting any non-zero value as `true`? It seems more error-prone IMO. If any value outside of `0` or `1` are used, it's probably a mistake.